### PR TITLE
Replaced single wrist with wrist differential.

### PR DESCRIPTION
**Pull request by AyushKulk**
Replaced instances of single wrist motor with wrist differentials for left and right gears and added logic to handle wrist pitch and roll control.

### CODE REVIEW

**Comment by abhaybd** [comment]
Just to write down what we talked about, let's hold off on merging before sim testing.

File: src/world_interface/kinematic_common_interface.cpp [hunk 1]
```diff
@@ -171,7 +171,14 @@ double getJointPowerValue(types::jointid_t joint) {
 
 void setJointMotorPower(robot::types::jointid_t joint, double power) {
 	using robot::types::jointid_t;
-	if (Constants::JOINT_MOTOR_MAP.find(joint) != Constants::JOINT_MOTOR_MAP.end()) {
+
+	if (joint == jointid_t::shoulder) {
+		setMotorPower(robot::types::motorid_t::wristDiffRight, power);
+	} else if (joint == jointid_t::forearm) {
+		setMotorPower(robot::types::motorid_t::wristDiffLeft, power);
+	}
```
Comment (abhaybd) [inline src/world_interface/kinematic_common_interface.cpp:179]: Fine for testing, remember to remove!

File: src/world_interface/kinematic_common_interface.cpp [hunk 2]
```diff
@@ -190,6 +197,16 @@ void setJointMotorPower(robot::types::jointid_t joint, double power) {
 					robot::getMotorPositionsRad(Constants::arm::IK_MOTORS).getData());
 			}
 		}
+	}
+	 else if (joint == jointid_t::wristPitch || joint == jointid_t::wristRoll) {
+		kinematics::DiffWristKinematics diffWristKinematics;
+		setJointPowerValue(joint, power);
+		kinematics::jointpos_t jointPwr2(getJointPowerValue(jointid_t::wristPitch),
+										 getJointPowerValue(jointid_t::wristRoll));
+		kinematics::gearpos_t gearPwr2 = diffWristKinematics.jointPowerToGearPower(jointPwr2);
```
Comment (abhaybd) [inline src/world_interface/kinematic_common_interface.cpp:206]: Why do these have 2 at the end of their names?

File: src/world_interface/real_world_constants.h [hunk 3]
```diff
@@ -83,8 +83,8 @@ constexpr auto potMotors = frozen::make_unordered_map<motorid_t, potparams_t>(
 	  {.adc_lo = 123, .mdeg_lo = -200 * 1000, .adc_hi = 456, .mdeg_hi = 200 * 1000}},
 	 {motorid_t::forearm,
 	  {.adc_lo = 1208, .mdeg_lo = -180 * 1000, .adc_hi = 841, .mdeg_hi = 180 * 1000}},
-	 {motorid_t::wrist,
-	  {.adc_lo = 123, .mdeg_lo = -100 * 1000, .adc_hi = 456, .mdeg_hi = 100 * 1000}},
+	 //{motorid_t::wrist,
+	 //{.adc_lo = 123, .mdeg_lo = -100 * 1000, .adc_hi = 456, .mdeg_hi = 100 * 1000}},
```
Comment (abhaybd) [inline src/world_interface/real_world_constants.h:87]: You can remove this line, and can you add entries for the wristDiffLeft and wristDiffRight? Just dummy values in the map is fine (all 0s) we'll have to calibrate it when it's mounted

File: src/world_interface/real_world_constants.h [hunk 4]
```diff
@@ -98,7 +98,6 @@ constexpr auto motorSerialIDMap = frozen::make_unordered_map<motorid_t, can::dev
 	 {motorid_t::shoulder, DEVICE_SERIAL_MOTOR_SHOULDER},
 	 {motorid_t::elbow, DEVICE_SERIAL_MOTOR_ELBOW},
 	 {motorid_t::forearm, DEVICE_SERIAL_MOTOR_FOREARM},
-	 {motorid_t::wrist, DEVICE_SERIAL_MOTOR_WRIST},
```
Comment (abhaybd) [inline src/world_interface/real_world_constants.h:101]: We need to add entries here for `wristDiffLeft` and `wristDiffRight`. Note sure about the exact serial ids though I'll have to get back to you. For now, put a dummy value (set both to `DEVICE_SERIAL_MOTOR_WRIST` and leave a TODO note)

**Review by abhaybd — CHANGES_REQUESTED** [review]
Great work on this! Just some minor notes, and once we get more info from firmware we can fill in the exact values for some stuff

File: src/world_interface/real_world_constants.h [hunk 5]
```diff
@@ -79,8 +79,10 @@ constexpr auto potMotors = frozen::make_unordered_map<motorid_t, potparams_t>({
 	 {.adc_lo = 123, .mdeg_lo = -200 * 1000, .adc_hi = 456, .mdeg_hi = 200 * 1000}},
 	{motorid_t::forearm,
 	 {.adc_lo = 1208, .mdeg_lo = -180 * 1000, .adc_hi = 841, .mdeg_hi = 180 * 1000}},
-	{motorid_t::wrist,
-	 {.adc_lo = 123, .mdeg_lo = -100 * 1000, .adc_hi = 456, .mdeg_hi = 100 * 1000}},
+	{motorid_t::wristDiffLeft,
+	 {.adc_lo = 0, .mdeg_lo = -100 * 0, .adc_hi = 0, .mdeg_hi = 100 * 0}},
+	{motorid_t::wristDiffRight,
+	 {.adc_lo = 0, .mdeg_lo = -100 * 0, .adc_hi = 0, .mdeg_hi = 100 * 0}},
```
Comment (Geeoon) [inline src/world_interface/real_world_constants.h:85]: Are these values placeholders?  If they are, you should leave a comment so you remember to add the real values.

**Review by Geeoon — CHANGES_REQUESTED** [review]
Just left a super small comment, looks good otherwise.

File: src/world_interface/real_world_constants.h (hunk 5 above)
Comment (AyushKulk) [reply src/world_interface/real_world_constants.h:85]: Yes there should be a comment there I must've forgotten when I moved everything to the updated master

**Review by abhaybd — APPROVED** [review]
Works as far as I can tell, electrical issues are hindering full testing but it is what it is.

**Pull request merged**